### PR TITLE
[MM-24321] server/metrics: remove globals

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
With this PR; I added all metrics related variables into `metrics` struct and passed it among required structs e.g. server etc.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24321

### CODE REVIEW

File: server/metrics.go [hunk 1]
```diff
@@ -24,181 +22,168 @@ const (
...
-func NewPrometheusHandler() http.Handler {
-	return promhttp.Handler()
+	return m
 }
 
-func incrementNotificationTotal(platform, pushType string) {
-	if MetricsEnabled {
-		metricNotificationsTotal.WithLabelValues(platform, pushType).Inc()
-	}
+func (m *metrics) shutdown() {
```
Comment (agnivade) [inline server/metrics.go:116]: Is this only for the tests?

File: server/metrics.go [hunk 2]
```diff
@@ -24,181 +22,168 @@ const (
...
+			Name: metricNotificationResponseName,
+			Help: "Notifiction request latency distribution"},
+			[]string{"platform"}),
+		metricServiceResponse: prometheus.NewHistogram(prometheus.HistogramOpts{
+			Name: metricServiceResponseName,
+			Help: "Request latency distribution",
+		}),
+	}
 
-func init() {
```
Comment (agnivade) [inline server/metrics.go:108]: Great stuff on removing init :+1:

File: server/metrics.go [hunk 3]
```diff
@@ -24,181 +22,168 @@ const (
...
-var metricAPNSResponse = prometheus.NewHistogram(prometheus.HistogramOpts{
-	Name: metricAPNSResponseName,
-	Help: "Request latency distribution",
-})
-
-var metricFCMResponse = prometheus.NewHistogram(prometheus.HistogramOpts{
-	Name: metricFCMResponseName,
-	Help: "Request latency distribution",
-})
+func NewPrometheusHandler() http.Handler {
```
Comment (agnivade) [inline server/metrics.go:25]: godocs please while we're at it :)

File: server/server.go [hunk 4]
```diff
@@ -94,11 +101,10 @@ func (s *Server) Start() {
 	metricCompatibleSendNotificationHandler := s.handleSendNotification
 	metricCompatibleAckNotificationHandler := s.handleAckNotification
 	if s.cfg.EnableMetrics {
-		MetricsEnabled = true
 		metrics := NewPrometheusHandler()
 		router.Handle("/metrics", metrics).Methods("GET")
-		metricCompatibleSendNotificationHandler = responseTimeMiddleware(s.handleSendNotification)
-		metricCompatibleAckNotificationHandler = responseTimeMiddleware(s.handleAckNotification)
+		metricCompatibleSendNotificationHandler = s.responseTimeMiddleware(s.handleSendNotification)
```
Comment (agnivade) [inline server/server.go:106]: :100:

File: server/metrics.go (hunk 1 above)
Comment (isacikgoz) [reply server/metrics.go:116]: Yes, but I can also include the `Server.Stop()`

Comment (agnivade) [reply server/metrics.go:116]: Ok, let's do that then.

**Comment by isacikgoz** [comment]
Actually a smoke test would be fine since all of the metrics are affected with this. I suppose it is safe to watch general metrics like `metricNotificationResponse`.

**Pull request merged**